dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateMemoryStore.cs

```diff
@@ -0,0 +1,516 @@
+            if (json["_additional"]!["vector"] != null)
+            {
+                IEnumerable<float> floats = json["_additional"]!["vector"]!.AsArray().Select(a => a!.GetValue<float>());
+                vector = new(floats);
+            }
+
+            string text = json["sk_text"]!.GetValue<string>();
+            string description = json["sk_description"]!.GetValue<string>();
+            string additionalMetadata = json["sk_additional_metadata"]!.GetValue<string>();
+            string key = json["sk_id"]!.GetValue<string>();
+            DateTime? timestamp = json["sk_timestamp"] != null
+                ? Convert.ToDateTime(json["sk_timestamp"]!.GetValue<string>(), CultureInfo.InvariantCulture)
+                : null;
+
+            MemoryRecord memoryRecord = MemoryRecord.LocalRecord(
+                id,
+                text,
+                description,
+                vector,
+                additionalMetadata,
+                key,
+                timestamp);
+
+            double distance = json["_additional"]!["distance"]!.GetValue<double>();
+
+            result.Add((memoryRecord, distance));
+        }
+
+        foreach ((MemoryRecord, double) kv in result)
+        {
+            yield return kv;
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<(MemoryRecord, double)?> GetNearestMatchAsync(
+        string collectionName,
+        Embedding<float> embedding,
+        double minRelevanceScore = 0,
+        bool withEmbedding = false,
+        CancellationToken cancellationToken = default)
+    {
+        IAsyncEnumerable<(MemoryRecord, double)> results = this.GetNearestMatchesAsync(
+            collectionName,
+            embedding,
+            minRelevanceScore: minRelevanceScore,
+            limit: 1,
+            withEmbeddings: withEmbedding,
+            cancellationToken: cancellationToken);
+
+        (MemoryRecord, double) record = await results.FirstOrDefaultAsync(cancellationToken).ConfigureAwait(false);
+
+        return (record.Item1, record.Item2);
+    }
+
+    // Get a class description, useful for checking name collisions
+    private static string ToWeaviateFriendlyClassDescription(string collectionName)
+    {
+        return $"Semantic Kernel memory store for collection: {collectionName}";
+    }
+
+    // Convert a collectionName to a valid Weaviate class name
+    private static string ToWeaviateFriendlyClassName(string collectionName)
+    {
+        // Prefix class names with SK to ensure proper case for Weaviate Classes
+        return $"SK{s_classNameRegEx.Replace(collectionName, string.Empty)}";
+    }
+
+    // Check to see if a class name is possibly a semantic kernel collection
+    private static bool IsPossibleSemanticKernelCollection(string collectionName)
+    {
+        if (!collectionName.StartsWith("SK", StringComparison.Ordinal))
+        {
+            return false;
+        }
+
+        string remainder = collectionName.Substring(2);
```

Comment:
If we move `SK` prefix to separate constant on class level, we can use `Constant.Length` and avoid magic number.